pandas/core/internals/ops.py

```diff
@@ -15,7 +20,9 @@
 )
 
 
-def _iter_block_pairs(left: "BlockManager", right: "BlockManager"):
+def _iter_block_pairs(
+    left: "BlockManager", right: "BlockManager"
+) -> Generator[BlockPairInfo, None, None]:
```

Comment:
Iterable (or Iterator) is adequate.

from https://mypy.readthedocs.io/en/stable/kinds_of_types.html#generators

>A basic generator that only yields values can be annotated as having a return type of either Iterator[YieldType] or Iterable[YieldType].